src/internal.c

```diff
@@ -22665,6 +22665,10 @@ int DecodePrivateKey(WOLFSSL *ssl, word16* length)
         WOLFSSL_START(WC_FUNC_CLIENT_HELLO_SEND);
         WOLFSSL_ENTER("SendClientHello");
 
+        if (ssl == NULL || ssl->arrays == NULL) {
+            return BAD_FUNC_ARG;
+        }
```

Comment:
Is there a case where ClientHello should be sent although `ssl->arrays` is NULL? E.g. when `ssl->options.connectState` is not `CONNECT_BEGIN` and DTLS is disabled. However don't know if that makes sense?

Reply:
Yes, potentially.  Updated this check.